pkg/diagnose/cni.go

```diff
@@ -0,0 +1,215 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package diagnose
+
+import (
+	"context"
+	"fmt"
+	"github.com/submariner-io/submariner-operator/internal/execute"
+	"github.com/submariner-io/submariner-operator/pkg/reporter"
+
+	"github.com/pkg/errors"
+	"github.com/submariner-io/submariner-operator/internal/cli"
+	operatorConstants "github.com/submariner-io/submariner-operator/internal/constants"
+	submv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/routeagent_driver/constants"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes"
+)
+
+var supportedNetworkPlugins = []string{
+	constants.NetworkPluginGeneric, constants.NetworkPluginCanalFlannel, constants.NetworkPluginWeaveNet,
+	constants.NetworkPluginOpenShiftSDN, constants.NetworkPluginOVNKubernetes, constants.NetworkPluginCalico,
+}
+
+var calicoGVR = schema.GroupVersionResource{
+	Group:    "crd.projectcalico.org",
+	Version:  "v1",
+	Resource: "ippools",
+}
+
+func CheckCNIConfig(cluster *execute.Cluster) bool {
```

Comment:
The public APIs in _pkg_ cannot reference anything in _internal_ as it wouldn't be accessible by a third party user outside the repo. I suggest to redefine `Cluster` in _pkg/cluster/info.go_ as `Info` so users reference it as `cluster.Info`. We'd then have to move the `execute` package under _cmd/subctl_ since _internal_ cannot reference _pkg_ (as that would create a circular dependency).

Reply:
Ditto